### DISCUSSION TITLE
How to use `WITH` clause (CTE) along `WITH FILL`

### DISCUSSION THREAD

**Discussion by sxm51** [ClickHouse/ClickHouse#50687] [General]
Hi Guys,

Was trying to create a query which has `WITH` clause (CTE) and `WITH FILL`.
The query appears to fail. What's the correct way to create a query for this.

Here is an example which fails:

```
WITH
  1 AS fillStep,
  2 AS multipier
SELECT number
FROM
(
    SELECT number * multipier AS number
    FROM system.numbers
    LIMIT 5
)
ORDER BY number ASC WITH FILL STEP fillStep
```

If I remove the top `WITH` clause, query works as expected.

```
SELECT number
FROM
(
    SELECT number * 2 AS number
    FROM system.numbers
    LIMIT 5
)
ORDER BY number ASC WITH FILL STEP 1
```

**Comment by tbragin** [comment]
Could you include the error message? And what version are you using?

@devcrafter You may know if this syntax is expected to be supported.

**Comment by sxm51** [comment]
Tried it with v23.3.x and v23.5.1 .

Missed to the add the error message.

Error:

```
Received exception:
Code: 47. DB::Exception: Missing columns: 'fillStep' while processing query: 'fillStep', required columns: 'fillStep' 'fillStep'. (UNKNOWN_IDENTIFIER)
```

**Comment by devcrafter** [comment]
This syntax is not supported right now, I'd consider it as a bug. But it's supported with new analyzer:
```
WITH
    1 AS fillStep,
    2 AS multipier
SELECT n
FROM
(
    SELECT number * multipier AS n
    FROM system.numbers
    LIMIT 5
)
ORDER BY n ASC WITH FILL STEP fillStep
SETTINGS allow_experimental_analyzer = 1

┌─n─┐
│ 0 │
│ 1 │
│ 2 │
│ 3 │
│ 4 │
│ 5 │
│ 6 │
│ 7 │
│ 8 │
└───┘
```
